test/integration/integration_test.go

```diff
@@ -81,8 +81,8 @@ func FeatureContext(s *godog.Suite) {
 
 	// steps to execute minishift commands
 	s.Step(`Minishift (?:has|should have) state "([^"]*)"`, minishift.shouldHaveState)
+	s.Step(`executing "minishift ([^"]*)" (succeeds|fails)$`, executingMinishiftCommandSucceedsOrFails)
```

Comment:
+1, this is already better than what we had. Explicitly list the potential states. I would have gone even a step further and added to dedicated steps. This would be even more explicit. Code wise we can still share some implementation details.